### DISCUSSION TITLE
Multiple languages maybe?

### DISCUSSION THREAD

**Discussion by martin6982** [seerge/g-helper#225] [Q&A] [answered]
Hi @seerge I come from Taiwan and I'm so happy to find this app on reddit, do you have a plan to make G-helper have mutiple languages? I think I can help the translation about the part of "Traditional Chinese" and maybe "Simplified Chinese". Maybe my coding experience is not good enough, but I can try my best.

also, I have a request that if I can introduce this app to the forum"https://www.gamer.com.tw/" which is the most popular gaming forum in Taiwan, because I found that G-Helper isn't well known between Mandarin users in Taiwan or China, we are also struggle in the issue about thermal, performance, and so on. I think this app will help Mandarin users a lot. Of course I will indicate you as the original author and share the app with your github link, please let me know if you are willing to agree that, thank you so much!

**Comment by seerge** [comment, marked as answer]
Hello @martin6982 , glad that you like the app.

Actually it has been just translated to Simplified Chinese - https://github.com/seerge/g-helper/releases/tag/v0.50 (feel free to try)

Also readme exists in Chinese too https://github.com/seerge/g-helper/blob/main/docs/README.zh-CN.md 

And sure - you can introduce the app to more users, just don't forget a link to this github page.

![Screenshot 2023-04-11 221528](https://user-images.githubusercontent.com/5920850/231278622-d95fedc6-b636-44d3-acf7-76cdbcd07577.png)

**Reply by martin6982** [reply]
thank you for replying me and I will introduce the app in these few days.
but after I got the v0.50, I can't find the option or guide to change the language,it's still in English, what should I do?

**Reply by seerge** [reply]
@martin6982 it takes language from system settings, i.e. if your windows is in english - it will be in english :)

**Reply by martin6982** [reply]
My Windows11 setting is in Chinese(Tradional Chinese,Taiwan https://imgur.com/yxzOEMQ ) already, but G-helper still in English, maybe Tradional Chinese is not recognized as Simplified Chinese I guess?

**Reply by seerge** [reply]
@martin6982 it could be, i can try to add a fix in a next build for that. Can you try to set windows to Simplified Chinese  ?

**Reply by seerge** [reply]
@martin6982 or actually - can you just try this build ?
[GHelper.zip](https://github.com/seerge/g-helper/files/11204953/GHelper.zip)

**Reply by martin6982** [reply]
> @martin6982 or actually - can you just try this build ? [GHelper.zip](https://github.com/seerge/g-helper/files/11204953/GHelper.zip)

it works!

**Reply by seerge** [reply]
@martin6982 ok, cool. I have just reuploaded it over existing https://github.com/seerge/g-helper/releases/tag/v0.50 - so people can use official link :) 

It will default all variants of system languages ``zh-CN, zh-TW, zh-hans, zh-hant`` to current Simplified Chinese translation

**Reply by martin6982** [reply]
but I still wish it will have Tradional Chinese, it would be perfect to me haha.

**Reply by martin6982** [reply]
> @martin6982 ok, cool. I have just reuploaded it over existing https://github.com/seerge/g-helper/releases/tag/v0.50 - so people can use official link :)
> 
> It will default all variants of system languages `zh-CN, zh-TW, zh-hans, zh-hant` to current Simplified Chinese translation

ok let me try, thank you so much!

**Reply by martin6982** [reply]
> @martin6982 ok, cool. I have just reuploaded it over existing https://github.com/seerge/g-helper/releases/tag/v0.50 - so people can use official link :)
> 
> It will default all variants of system languages `zh-CN, zh-TW, zh-hans, zh-hant` to current Simplified Chinese translation

@seerge 
I found the language file(Strings.zh.resx) and entirely translated it into Tradional Chinese by myself, but I don't know how to modify my own G-helper, can you update it?
Here is the zh-tw file: https://drive.google.com/file/d/1Cq-UKZEQQaZeCvU5ZsdE8E3y3ZlU7_72/view

**Reply by seerge** [reply]
[GHelper.zip](https://github.com/seerge/g-helper/files/11209967/GHelper.zip)

can you check if this build shows correct translation for you ?

**Reply by martin6982** [reply]
> [GHelper.zip](https://github.com/seerge/g-helper/files/11209967/GHelper.zip)
> 
> can you check if this build shows correct translation for you ?

The partition of my translation is correct now, but I found that I miss the [option list](https://imgur.com/uFeh5Mc) of the CPU boost, where can I found it and translate it?

**Reply by seerge** [reply]
> The partition of my translation is correct now, but I found that I miss the [option list](https://imgur.com/uFeh5Mc) of the CPU boost, where can I found it and translate it?

It's part of windows power plan setting. I think it's better to keep setting names exactly as they are used there - so it's easier to google / find relative info about them

**Reply by martin6982** [reply]
And I found a small mistake in my translation and I already fixed it, maybe you can update it in the next version
it's line 382 "<value>切換Aura</value>"
or use this updated file https://drive.google.com/file/d/18V9xk-9AYKXIQKJUSIsOnrzUQr_Ixay8/view?usp=share_link

**Reply by seerge** [reply]
Ok

**Reply by seerge** [reply]
@martin6982 pushed live in https://github.com/seerge/g-helper/releases/tag/v0.52

**Reply by martin6982** [reply]
great!

**Reply by martin6982** [reply]
Hi @seerge, I just finished my post on the forum I mentioned before, though you might not speak Chinese, It's still my pleasure to share with you! here is the link : https://forum.gamer.com.tw/C.php?bsn=60030&snA=622942

**Reply by seerge** [reply]
@martin6982 that's a long post! thanks for the effort

**Comment by martin6982** [comment]
Hi @seerge I just updated the "Strings.zh-TW" file on the github code editor page, but I am not sure if  you can see that(It's my first time using the function), can you check that out?

**Reply by seerge** [reply]
@martin6982 that's not how it works. In your case it would be easier to download latest version of strings-tw, edit it and upload it in this thread :)

**Reply by martin6982** [reply]
ok, then here is the file which I updated:
https://drive.google.com/file/d/14zU0TscEDwmKyUhK71yhGCEIUH27WeNI/view?usp=share_link
thank you!

**Reply by martin6982** [reply]
@seerge I updated the new translation based on v0.64, here is the file:
https://drive.google.com/file/d/1ySDs-0XeIY11IsEvu4KkWBw0JrI7kAAa/view?usp=sharing
thank you!

**Reply by seerge** [reply]
Oke, thanks. I will include it in next releases

**Reply by martin6982** [reply]
@seerge here is the v0.65 translation:
https://drive.google.com/file/d/1_5lruXgwM7sY8bc6E832UT7ohBSjPtly/view?usp=share_link
thanks!

**Reply by seerge** [reply]
@martin6982 oke

**Reply by martin6982** [reply]
@seerge here is the v0.70 translation:
https://drive.google.com/file/d/1oDtJJnlhu1n4FISvInzI5eDGLYv826nF/view?usp=drive_link
thanks!

**Reply by martin6982** [reply]
@seerge here is the v0.75 translation:
https://drive.google.com/file/d/1OD_QnQD7sB-k0YOf2s8Q8kFGP96-YJ8k/view?usp=drive_link
thanks!

**Reply by martin6982** [reply]
@seerge here is the v0.78 translation:
https://drive.google.com/file/d/1nZxKnaYQ1WyxLj2ukEdrKrMoTg-1yjiQ/view?usp=drive_link
thanks!

**Reply by martin6982** [reply]
@seerge here is the v0.79 translation:
https://drive.google.com/file/d/1dvvxyOpM9FYIJ4FXz4X1xtlzmoX2YjtR/view?usp=drive_link
thanks!

**Reply by martin6982** [reply]
@seerge here is the v0.82 translation:
https://drive.google.com/file/d/1tBiXiXLitTLMuQqTP2Zm9GbkooZb2xZI/view?usp=drive_link
thanks!

**Comment by richardotomislav** [comment]
I'm using the app, and it's taking the Windows language, as you said on this thread. However, is there another way of setting the app language without changing the entire OS language?

**Reply by seerge** [reply]
No, currently it takes language from system locale on startup. Why would you want different language from your system ?

**Reply by richardotomislav** [reply]
Because the Spanish translation is basically "useless", it's very bad as it was done with a translator. For example, "Quit" is translated as "Quitar" which should be translated as "Exit". 

I would help with the translation if you point to me some directions to where the files are located in the code. But for now, I'm using Windows on English.

**Reply by seerge** [reply]
It's not done with translator, @marcelomijas helped here.

**Reply by marcelomijas** [reply]
Hi. @JesusJMUJI I am sorry that the translation might be bad for you. The translation hasn't been done with a translator. "Exit" and "Quit" are two different things. I leave "Quit" as "Quitar" because it definitely "quits" the pogram (closes completely). "Exit" as "Salir" could be seen as closing the window for someone. I tried to leave the translation with the closest meaning to the original English without exceeding on words (that is an important one). Can you provide more examples of bad translation?

Also, I don't know if you are spanish from Spain or from Latin America. Note that some words look better on Latin Spanish than on Castillian Spanish or vice versa...

I am a native spanish, currently a Professor at an uni in Spain, teaching Economics in English.... Additionally, I had some other work and project experiences in English. That is the reason that I felt confident to do this translation.

**Reply by richardotomislav** [reply]
My bad then, I didn't say it with bad intentions, I could help with the translations if you give me some directions as I said. 

But a language switch would be appreciated.

**Reply by richardotomislav** [reply]
> Hi. @JesusJMUJI I am sorry that the translation might be bad for you. The translation hasn't been done with a translator. "Exit" and "Quit" are two different things. I leave "Quit" as "Quitar" because it definitely "quits" the pogram (closes completely). "Exit" as "Salir" could be seen as closing the window for someone. I tried to leave the translation with the closest meaning to the original English without exceeding on words (that is an important one). Can you provide more examples of bad translation?
> 
> Also, I don't know if you are spanish from Spain or from Latin America. Note that some words look better on Latin Spanish than on Castillian Spanish or vice versa...
> 
> I am a native spanish, currently a Professor at an uni in Spain, teaching Economics in Enghish.... Additionally, I had some other work and project experiences in English. That is the reason that I felt confident to do this translation.


I speak Castillian Spanish, what you have done is called a "false friend" as you might know. The right translation is "Salir" as you are "exiting" the program.

I haven't been able to check the program to its fullest, so I cannot say if there are any more mistranslations. 

Still, I commend you for the labor of translating it even if you are not a native. My English isn't perfect either. 

_Also, this is very off-topic, but it would be a coincidence if we were at the same university._

**Reply by marcelomijas** [reply]
No offenses taken :)

As I said, because my experience I felt confident about translating, but I am aware that I can have mistakes too. Can you provide more examples of bad translation? I could be wrong and/or it might be room for improvement...

**Reply by seerge** [reply]
Just in case https://github.com/seerge/g-helper/blob/main/app/Properties/Strings.es.resx this is a file containing all strings for spanish

**Reply by richardotomislav** [reply]
> Just in case https://github.com/seerge/g-helper/blob/main/app/Properties/Strings.es.resx this is a file containing all strings for spanish

Many thanks, I'll take a look when I have some free time.

**Reply by richardotomislav** [reply]
> No, currently it takes language from system locale on startup. Why would you want different language from your system ?

One more question @seerge, would be a problem if I open an issue about this? So, it can be considered as a new feature?

**Comment by marcelomijas** [comment]
@JesusJMUJI I disagree in this case... at the beginning I also thinked about translating it to "Salir", but again, I know a lot of people that this might be confusing. I understand that Exit a program is closing the window (but it might be running in the background), but quitting (Quitar) is closing it completely.

About the off topic: URJC or Complutense? (I teach officially on the first and help a group on the second) XD

**Reply by richardotomislav** [reply]
"Quitar" in this case you are saying that are "getting it out of the way", like moving a chair because it's blocking your path. 

"Salir" would be correct in this case because you are "leaving" the program, or "closing" it in the sense of "I don't want to use anymore" and when you "close" a window you are, well, closing it, you want it out of the way. Like in the chair example. 

"Cerrar" would be the appropriate translation when closing the program and "Salir" when you are stopping the process entirely (right click in task bar).

_And about the off-topic, nah you are on another region. I'm glad you are teaching in Spain, good luck with your endeavours._

**Reply by seerge** [reply]
@JesusJMUJI  @marcelomijas  I see a PR https://github.com/seerge/g-helper/pull/349, but since I'm not spanish i see lots of changes, and not sure if I should merge it or not ? :|  (based on discussion here)

**Reply by seerge** [reply]
PS I can imagine Marcelo deliberately shortened some labels as they are hard to fit into ui (which is good)

**Reply by marcelomijas** [reply]
@seerge Yeah, that was the most complicated part, the shortening of labels. @JesusJMUJI I am writing comments on the PR. Some things are ok, like changing "Perfil" to "Curva", but be aware that Armoury Crate uses the word "Perfil" for the custom fans tool (at least in the 2020 G14)... but it is ok. :)

In contrast, other things are... well, look at the comments...

**Reply by seerge** [reply]
@marcelomijas and right - generally rule of thumb should be calling things same as they are called in AC ;) to simplify migration

**Reply by richardotomislav** [reply]
Sure thing, I'm cancelling the pull request for now. I need more experience before doing any of this, or at least asking beforehand. 
I'm thankful for the input, too.

**Reply by marcelomijas** [reply]
I made all the comments @JesusJMUJI . Thank you!

@seerge I also see that the Spanish file has changed a little bit with some new field additions, Number of rows doesn't match with the English Strings.resx I will try to make the Spanish one complete today at night.

**Reply by seerge** [reply]
@marcelomijas yeah, mainly cause of added checkboxes in Extra (for Strix models) and some other stuff, i just don't want to bother translators every time :)

**Comment by TomBonnot** [comment]
Might be a weird question but : 
- Where do I change language ? I want to use GHelper with my own traduction hehe 
- Since I have done a fork of your project, will I have to do another fork later if I want to change new strings ? It’s my first time using fork haha so I am kinda curious ! 
Anyway, that is cool, I think I will work on a fr and pt translation about the README(if it is useful, since I guess everyone can understand English)

**Reply by seerge** [reply]
@TomBonnot only in your system, app will pick it up. If you are debugging via visual studio -> also here https://github.com/seerge/g-helper/blob/020a7d074fd6501fec5688eb2e0077076debf388/app/Program.cs#LL45C19-L45C19 (uncomment line and set language)

You can either update your existing fork or create a new one, it doesn't matter

I wouldn't rush with translating README, as it changes :) + it's much bigger than any set of labels in the app
